### DISCUSSION TITLE
Netbox-GLPI intergration

### DISCUSSION THREAD

**Discussion by compgeniuses** [netbox-community/netbox#8977] [Ideas]
Is anyone currently working on any **Netbox** and **GLPI** intergation,especially with the new GLPI10 integration. Netbox and glpi are both superb open source solution for infrastructure documention.
The both could have fantastic integrations with the new Netbox module system
**GLPI is found here**: https://github.com/glpi-project/glpi

specifically, the Domains management and DNS records availed in glpi,

**Comment by jon-nfc** [comment]
_first result in the search engine!_

G'day @compgeniuses, did you end up finding anything? As I'm looking for the same, although nothing can be found so likely I'll have to create something.

**Reply by compgeniuses** [reply]
nope, i never found any such solution. if you have the capacity to build or sponsor such an endeavour i am sure many would highly appreciate

**Reply by jon-nfc** [reply]
Unfortunate, as GLPI is now very usable due to the significant improvements over the last few years.

Yes I do have the capacity to undertake such an endeavour. I've just started a [project in my repo](https://gitlab.com/nofusscomputing/projects/ansible/collections/netbox) so that I can start to integrate netbox with GLPI.

What direction this project will take at this stage is too early to tell, although I can say my thoughts are that netboox will be the single source of truth and the first part of the integration will be to add the netbox items to GLPI (and keep them in sync) so as to take advantage of the ITAM that GLPI offers.

**Reply by compgeniuses** [reply]
you need to understand that netbox handles well matters network management and glpi handles well matters IT assets geenrally.

so you can blanket it that netbox becomes the single source of truth, as other areas GLPI is more robust and steonger.

Glpi has some IPAM but its not that great compared to what netbox handles, Maybe that would be the place to start from.

Implement the ability for GLPI to source and link netbox objext, as well as NEtbox being able to fetch and link glpi objects.

That way, no matter where the Objects were added, its possible to interlink.

Equally, you need o take a deeper look at that is stronger in which areas and objects, that way, youy have users using one for some types of Objects, and the other for  others

For servers and Virtual machines etc, netbox does it great. so the ability to have these kindly of objects interlinked, if i ad a server in glpi or netobx it also apprears on the other one, then the addidional information and drill down could be handled by either

**Reply by jon-nfc** [reply]
> you need to understand that netbox handles well matters network management and glpi handles well matters IT assets geenrally.
> so you can blanket it that netbox becomes the single source of truth, as other areas GLPI is more robust and steonger.
> 

Yes indeed. IT Asset Management (ITAM) is most definitely GLPI.

> Glpi has some IPAM but its not that great compared to what netbox handles, Maybe that would be the place to start from.

What GLPI has IPAM wise I don't think I'd call it that. It's all over the place and missing anything to complete the config of such.

> Implement the ability for GLPI to source and link netbox objext, as well as NEtbox being able to fetch and link glpi objects.
> 
> That way, no matter where the Objects were added, its possible to interlink.

I have considered this, However two way sync is not yet a direction I'm going to take. Netbox to GLPI simple enough however GLPI to  Netbox is a different kettle of fish. If this feature is ever added, It's going to take significant planning to link the different fields.

Project is well under way - https://gitlab.com/nofusscomputing/projects/ansible/collections/netbox/-/merge_requests/1 with Netbox to GLPI sync for computers (Device and Virtual Machine) and network equipment done and now working on to the selected entity.

**Reply by compgeniuses** [reply]
I already hve an easy way to deploy GLPI here: https://github.com/geniusdynamics/ns8-glpi as a Nethserver 8 Module, i think with what you're working on, its time for me to work on a Netbox Module as well. Ill target v4 and wait till its released for a final release in this case, makes it easier for me. then i can hep test the integration, and if there is any more works to be done with my installation to support this integration, ill work on the same as well.

**Comment by compgeniuses** [comment]
So in relation to your one way sync of Netbox to GLPI, will Data centers be added into NEtbox, as opposed to Glpi?
Appliances, Clusters

What would be the situation handling for those added into Glpi, will it be. if object added into Glpi, then not required in netbox, but if added into netbox, then required in Glpi as well.

**Reply by jon-nfc** [reply]
> So in relation to your one way sync of Netbox to GLPI, will Data centers be added into NEtbox, as opposed to Glpi? Appliances, Clusters

I haven't noticed any data centers in Netbox, unless I'm blind. 

> What would be the situation handling for those added into Glpi, will it be. if object added into Glpi, then not required in netbox, but if added into netbox, then required in Glpi as well.

The direction I'm taking is to sync from NetBox to GLPI just what is required to support the GLPI feature that NetBox is lacking. for example, to utilize the ITAM features of GLPI, name, asset tag, serial number location etc is required. If the feature is in GLPI and not Netbox, then there's nothing to be done. Where there's overlap are the areas to target.

As for existing items in GLPI, currently you can populate the GLPI ID within the Netbox Item, However this will eventually change to sync items with matching Serial numbers.

**Reply by teadur** [reply]
@jon-nfc  They seem to be available now:
https://glpi-user-documentation.readthedocs.io/fr/latest/modules/management/data-centers.html
